### ISSUE TITLE
Text label with zoom stops behaves incorrectly on Android

### ISSUE THREAD

**Issue by clydebarrow** [mapbox/mapbox-gl-native#6303]
**Platform: Android:**
**Mapbox SDK version 4.2.0-beta.2:**

### Steps to trigger behavior

 1. I have a style that changes labels for a point based on zoom level:

                "text-field": {
                    "base": 1,
                    "stops": [
                        [
                            0,
                            "{code}"
                        ],
                        [
                            10,
                            "{name}"
                        ],
                        [
                            22,
                            "{name}"
                        ]
                    ]
                },
Previewing this map in Studio, or using the sharing URL gives the expected behaviour, i.e. the label displays the {code} property at low zoom levels, and the {name} property at high zoom levels: 
![screen shot 2016-09-10 at 8 48 39 pm](https://cloud.githubusercontent.com/assets/2366188/18410538/40e70d4e-77a9-11e6-8a35-76f8fcfd5f9d.png)
. When using this map in an Android app only the higher zoom label ({name} in the above example) is displayed, irrespective of the zoom level.
![screenshot_2016-09-10-20-43-26](https://cloud.githubusercontent.com/assets/2366188/18410544/5adc0088-77a9-11e6-8b17-e6c15e08f32f.png)

**Comment by clydebarrow** [comment]
After some experimentation I found that adding an extra stop to the function makes it work on Android. Changing the above example to:

```xml
                "text-field": {
                    "base": 1,
                    "stops": [
                        [
                            0,
                            "{code}"
                        ],
                        [
                            9,
                            "{code}"
                        ],
                        [
                            10,
                            "{name}"
                        ],
                        [
                            22,
                            "{name}"
                        ]
                    ]
                }
```

results in {code} being shown up to zoom level 10.

**Comment by jfirebaugh** [comment]
Possibly same root cause as #5823.

**Comment by clydebarrow** [comment]
This begs the question as to what is the expected behaviour. With numeric functions it's possible to interpolate, but where the values are discrete there needs to be some other algorithm. Is this specified anywhere? If not I propose the following for functions with discrete values:

```
Select the value from the nearest zoom stop with a value less than or equal to the current zoom;
If no such stop exists, select the value from the nearest zoom stop with a value greater than the current zoom.
If no such stop exists the function is invalid.
```

This shouldn't break anything existing, but will make my original example and that from #5823 work as expected.

**Comment by clydebarrow** [comment]
Ok, reading the mapbox-gl-native code, it appears the current algorithm chooses the higher zoom stop if it exists, i.e. the reverse of my proposal. That's a reasonable implementation (if documented) but it does not behave in the same manner as the Studio preview. One or the other needs to change. I tried looking at the mapbox-gl-js code to see what it does, but I don't speak Javascript so I couldn't figure it out.

**Comment by jfirebaugh** [comment]
Fixed in #6328.
